### PR TITLE
ci: move Flagships to GH

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

1. include dispatch for flexible testing
2. simple customization to run on staging & prod
3. we do not expect that contributors would run these apps

see: https://github.com/Lightning-AI/lightning/pull/16392#discussion_r1072399251
followup of #16304 #16391 #16392 

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

File: .github/checkgroup.yml [hunk 1]
```diff
@@ -263,13 +264,21 @@ subprojects:
       - "app-examples (windows-2022, app, 3.9, oldest)"
       - "app-examples (windows-2022, lightning, 3.9, latest)"
 
+  - id: "lightning: Flagships"
+    paths:
+      - ".github/workflows/ci-flagship-apps.yml"
```
Comment (carmocca) [inline .github/checkgroup.yml:269]: Add `.github/actions/prep-apps/action.yml`?

**Review by carmocca — APPROVED** [review]

File: .github/checkgroup.yml (hunk 1 above)
Comment (Borda) [reply .github/checkgroup.yml:269]: I would keep it light as the only command we use here is already tested on several other places... as required :chipmunk:

File: .github/workflows/ci-examples-app.yml [hunk 2]
```diff
@@ -12,6 +12,7 @@ on:
       - ".github/workflows/ci-examples-app.yml"
       - "src/lightning_app/**"
       - "tests/integrations_app/**"
+      - "!tests/integrations_app/flagship/**"
```
Comment (carmocca) [inline .github/workflows/ci-examples-app.yml:15]: @Borda I added these exclusions to match the checkgroup config

**Review by awaelchli — APPROVED** [review]

File: .github/workflows/ci-examples-app.yml (hunk 2 above)
Comment (Borda) [reply .github/workflows/ci-examples-app.yml:15]: Yes, good ones, thx 🐰

**Review by akihironitta — APPROVED** [review]

**Pull request merged**